src/process.cr

```diff
@@ -513,6 +513,150 @@ class Process
     io.close if io
   end
 
+  # Returns the **real** user id of the current process.
+  def self.user_id : LibC::UidT
+    LibC.getuid
+  end
+
+  # Returns the **effective** user id of the current process.
+  def self.effective_user_id : LibC::UidT
+    LibC.geteuid
+  end
+
+  # Returns the **real** group id of the current process.
+  def self.group_id : LibC::GidT
+    LibC.getgid
+  end
+
+  # Returns the **effective** group id of the current process.
+  def self.effective_group_id : LibC::GidT
+    LibC.getegid
+  end
+
+  # :nodoc:
+  UID_DEFAULT = if LibC::UidT.new(0).is_a?(Int::Signed)
+                  LibC::UidT.new(-1)
+                else
+                  LibC::UidT::MAX
+                end
+  GID_DEFAULT = if LibC::GidT.new(0).is_a?(Int::Signed)
+                  LibC::GidT.new(-1)
+                else
+                  LibC::GidT::MAX
+                end
+
+  # Permanently transition to another account.
+  #
+  # user_id's, group_id's and groups are changed to the account provided.
+  #
+  # Call `chroot` or other privileged operations becore calling this method.
+  #
+  # Example:
+  #
+  # ```
+  # user = System::User.find_by name: "crystal"
+  # Process.become user
+  #
+  # user = System::User.find_by name: "crystal"
+  # group = System::Group.find_by name: "wheel"
+  # Process.become user, group # Use a different group other than the user's.
+  # ```
+  def self.become(user : System::User, group : System::Group? = nil)
+    group ||= user.group
+    # TODO: Call setgroups() when available using user.group_ids when available.
+    become_group group.id.to_i
+    become_user user.id.to_i
+  end
+
+  # Attempts to change real, effective, and saved user id's of the current process.
+  # When uid != 0 this is a one way transition.
+  #
+  # Example:
+  #
+  # ```
+  # Process.become_user 0 # Changes real, effective, saved user id to 0 (root).
+  # ```
+  def self.become_user(uid : LibC::UidT)
+    become_user ruid: uid, euid: uid, suid: uid
+  end
+
+  # Attempts to change real, effective, and/or saved user id's of the current process.
+  # Uid's not supplied in arguments are kept at their current values (if supported).
+  # Explicit setting of saved id's is not supported on all platforms.
+  #
+  # * *ruid*: real user id.
+  # * *euid*: effective user id.
+  # * *suid*: saved user id.
+  #
+  # Example:
+  #
+  # ```
+  # Process.become_user ruid: 1004, euid: 0 # Changes real and saved user id to 1004 and effective to 0 (root)
+  # Process.become_user euid: 0             # Only changes the euid.
+  # Process.become_user suid: 0             # Platform specific and may not function.
+  # ```
+  def self.become_user(*, ruid : LibC::UidT = UID_DEFAULT, euid : LibC::UidT = UID_DEFAULT, suid : LibC::UidT = ruid)
```

Comment:
I propose to name these arguments simply `real`, `effective` and `saved` instead of cryptic abbreviations that require an explanation.